### DISCUSSION TITLE
Do we have the resume after power loss feature?

### DISCUSSION THREAD

**Discussion by johncotten** [xblax/flashforge_ad5m_klipper_mod#77] [Q&A]
Not sure if that is supposed to work or not. I couldn't get a resume from pause working, so I tried an emergency stop. Did not get a resume option.

**Comment by xblax** [comment]
Afaik that is not even supported by Klipper. In general that's quite complicated as it would need a UPS or at least some kind of power loss signal that allows controlled pause of the print before the actual power-loss kicks in. But even manual pause/resume state does not survive Klipper restarts. 

Best recommendation for unstable power-grid would be to get a UPS with large enough battery to allow finishing the print.

**Reply by johncotten** [reply]
The stock software has power loss recovery.  My Sonic Pad has it too.Both running Klipper. I imagine it’s got to write the last completed gcode line to a temp file. Not sure it would work for ABS or ASA, where it would lose bed adhesion quickly. I’ll see if I can find any code examples.

**Comment by johncotten** [comment]
@xblax Don't get me wrong, you have done an amazing job with this project! I'm truly amazed by the work you and others have done to make this "beta" pretty amazing!  
It might not be a Klipper code, maybe python? Surely it's got to be in the orig_root files. This would definitely falls into the  enhancement category, but would be very helpful. It's not that I'm looking for power loss because of a power grid issue, but those times where maybe you need to hit emergency stop, or some other glitch that you need to turn the whole thing off. Ultimately it's possible if the plate cools too much the print could lose adhesion anyways.
